kong/tracing/propagation.lua

```diff
@@ -397,6 +411,32 @@ local function parse_aws_headers(aws_header)
   return trace_id, span_id, should_sample
 end
 
+local function parse_gcp_headers(gcp_header)
+  local warn = kong.log.warn
+
+  if type(gcp_header) ~= "string" then
+    return nil, nil, nil
+  end
+
+  local match, err = ngx.re.match(gcp_header, GCP_TRACECONTEXT_REGEX)
```

Comment:
```suggestion
  local match, err = ngx.re.match(gcp_header, GCP_TRACECONTEXT_REGEX, 'jo')
```
For better performance, see the options section of [ngx.re.match](https://github.com/openresty/lua-nginx-module#ngxrematch)